src/parseInlineTags.js

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * @param {RegExpMatchArray} match An inline tag regexp match.
+ * @returns {string}
+ */
+function determineTextStyle (match) {
+  const {separator, text} = match.groups;
+  const [, textEnd] = match.indices.groups.text;
+  const [tagStart] = match.indices.groups.tag;
+  if (!text) {
+    return 'plain';
+  } else if (separator === '|') {
+    return 'pipe';
+  } else if (textEnd < tagStart) {
+    return 'prefix';
+  }
+  return 'space';
+}
+
+/**
+ * Extracts inline tags from a description.
+ * @param {string} description
+ * @returns {InlineTag[]} Array of inline tags from the description.
+ */
+function parseDescription (description) {
+  const result = [];
+
+  // This could have been expressed in a single pattern,
+  // but having two avoids a potentially exponential time regex.
+
+  // eslint-disable-next-line prefer-regex-literals -- Need 'd' (indices) flag
+  const prefixedTextPattern = new RegExp(/(?:\[(?<text>[^\]]+)\])\{@(?<tag>[^}\s]+)\s?(?<namepathOrURL>[^}\s|]*)\}/gu, 'gud');
+  // The pattern used to match for text after tag uses a negative lookbehind
+  // on the ']' char to avoid matching the prefixed case too.
+  // eslint-disable-next-line prefer-regex-literals -- Need 'd' (indices) flag
+  const suffixedAfterPattern = new RegExp(/(?<!\])\{@(?<tag>[^}\s]+)\s?(?<namepathOrURL>[^}\s|]*)(?<separator>[\s|])?(?<text>[^}]*)\}/gu, 'gud');
+
+  const matches = [
+    ...description.matchAll(prefixedTextPattern),
+    ...description.matchAll(suffixedAfterPattern)
+  ];
+
+  for (const match of matches) {
+    const {tag, namepathOrURL, text} = match.groups;
+    const [start, end] = match.indices[0];
+    const textStyle = determineTextStyle(match);
+
+    result.push({
+      tag,
+      namepathOrURL,
+      text,
+      textStyle,
```

Comment:
@brettz9 Although your original proposal was to use `linkStyle` I think `textStyle` is more suitable since it actually describes the position of the text more than any link / url. Also "link" seems incorrect since this applies to parsing of any inlined tag.

Reply:
I see what you're saying, but the only other inline tag, `@tutorial`, takes a similar approach in its building its own form of "links". I think `textStyle` may be confusable with colors and such. Maybe `linkFormat` would be better, but I do think it relates to the nature of forming a link. I'm open to further suggestions, but that's my thinking right now.

Reply:
How about simply "style", "type" or "format"? I think neither would be ambitious in the context of an inline tag.

Reply:
Ok, `format` sounds good. Keeps a similar meaning for tutorial then too (rather than suggesting there is a certain kind/type of tutorial).

Reply:
Done 👍